projects/dsp-ui/src/lib/core/session.service.spec.ts

```diff
@@ -0,0 +1,106 @@
+
+    const dspConnSpy = {
+        admin: {
+            usersEndpoint: jasmine.createSpyObj('usersEndpoint', ['getUser'])
+        }
+    };
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [],
+            providers: [
+                {
+                    provide: DspApiConfigToken,
+                    useValue: dspConfSpy
+                },
+                {
+                    provide: DspApiConnectionToken,
+                    useValue: dspConnSpy
+                }
+            ]
+        });
+        service = TestBed.inject(SessionService);
+    }));
+
+    // mock localStorage
+    beforeEach(() => {
+        let store = {};
+
+        spyOn(localStorage, 'getItem').and.callFake(
+            (key: string): string => {
+                return store[key] || null;
+            }
+        );
+        spyOn(localStorage, 'removeItem').and.callFake(
+            (key: string): void => {
+                delete store[key];
+            }
+        );
+        spyOn(localStorage, 'setItem').and.callFake(
+            (key: string, value: string): string => {
+                return (store[key] = <any>value);
+            }
+        );
+        spyOn(localStorage, 'clear').and.callFake(() => {
+            store = {};
+        });
+    });
+
+    beforeEach(() => {
+
+        const valuesSpy = TestBed.inject(DspApiConnectionToken);
+
+        (valuesSpy.admin.usersEndpoint as jasmine.SpyObj<UsersEndpointAdmin>).getUser.and.callFake(
+            () => {
+                const loggedInUser = MockUsers.mockUser();
+                return of(loggedInUser);
+            }
+        );
+    });
+
+    it('should be created', () => {
+        expect(service).toBeTruthy();
+    });
+
+    it('should store user information in local storage', () => {
+
+        const userSpy = TestBed.inject(DspApiConnectionToken);
+        let ls: Session;
+
+        service.setSession(undefined, 'anything.user01', 'username');
+        expect(userSpy.admin.usersEndpoint.getUser).toHaveBeenCalledTimes(1);
+        ls = JSON.parse(localStorage.getItem('session'));
+        expect(ls.user.name).toEqual('anything.user01');
+        expect(ls.user.lang).toEqual('de');
+        expect(ls.user.sysAdmin).toEqual(false);
+        expect(ls.user.projectAdmin.length).toEqual(0);
+
+    });
+
+    it('should get the session with user information', () => {
+        service.setSession(undefined, 'anything.user01', 'username');
+
+        const session: Session = service.getSession();
+        expect(session.user.name).toEqual('anything.user01');
+        expect(session.user.lang).toEqual('de');
+        expect(session.user.sysAdmin).toEqual(false);
+        expect(session.user.projectAdmin.length).toEqual(0);
+    });
+
+    it('should destroy the session', () => {
+        service.destroySession();
+        const ls: Session = JSON.parse(localStorage.getItem('session'));
+        expect(ls).toEqual(null);
+    });
+
```

Comment:
I think it would also be a good idea to test the validateSession and updateSession methods. Flavie mentioned there was an issue with testing the validateSession?

Reply:
I don't know how to test validateSession. How can we get a mocked jwt from fake api? I have no idea... I'm open for suggestions.

Reply:
@kilchenmann you try something like this?! https://stackoverflow.com/questions/41114618/trying-to-mock-angular2-token-in-test 
It is a bit old though...

Reply:
@kilchenmann @flavens I think this is good, we just need some tests that check: 

- if isSessionValid returns false when there is no session
- if isSessionValid returns true when the internal dsp-ui session is still valid
- if isSessionValid calls `dspApiConnection.v2.auth.checkCredentials()` when the dsp-ui session is expired
    - For this, we will also need to mock the auth service. I don't know how to mock 2 services in the same test since they are both used for the `DspApiConnectionToken` in the providers

Reply:
I will try to write something...